runtime/src/main/java/mb/pie/runtime/taskdefs/CompositeTaskDefs.java

```diff
@@ -4,29 +4,35 @@
 import mb.pie.api.TaskDefs;
 import org.checkerframework.checker.nullness.qual.Nullable;
 
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.Objects;
+
 /**
  * Task definitions from parent and child task definitions.
  */
 public class CompositeTaskDefs implements TaskDefs {
-    private final TaskDefs parentTaskDefs;
+    private final List<TaskDefs> parentTaskDefs;
     private final TaskDefs childTaskDefs;
 
-    public CompositeTaskDefs(TaskDefs parentTaskDefs, TaskDefs childTaskDefs) {
-        this.parentTaskDefs = parentTaskDefs;
+    public CompositeTaskDefs(Collection<TaskDefs> parentTaskDefs, TaskDefs childTaskDefs) {
+        this.parentTaskDefs = new ArrayList<>(parentTaskDefs);
         this.childTaskDefs = childTaskDefs;
     }
 
+    @SuppressWarnings({"rawtypes", "unchecked"})
     @Override public @Nullable TaskDef<?, ?> getTaskDef(String id) {
         // Check parent first, such that child task definitions cannot override those from the parent.
-        final @Nullable TaskDef<?, ?> taskDef = parentTaskDefs.getTaskDef(id);
-        if(taskDef != null) {
-            return taskDef;
-        } else {
-            return childTaskDefs.getTaskDef(id);
-        }
+        return parentTaskDefs
+            .stream()
+            .map(taskDefs -> taskDefs.getTaskDef(id))
+            .filter(Objects::nonNull)
+            .findFirst()
+            .orElseGet(() -> (TaskDef) childTaskDefs.getTaskDef(id));
```

Comment:
Why the cast to `TaskDef` here?

Reply:
Thats due to a nasty Java type system quirck. When I remove it, or when I cast it to `TaskDef<?, ?>`, it gives this error:
```
devenv/pie/runtime/src/main/java/mb/pie/runtime/taskdefs/CompositeTaskDefs.java:32: error: incompatible types: bad return type in lambda expression
            .orElseGet(() -> childTaskDefs.getTaskDef(id));
                                                     ^
    TaskDef<CAP#1,CAP#2> cannot be converted to TaskDef<CAP#3,CAP#4>
  where CAP#1,CAP#2,CAP#3,CAP#4 are fresh type-variables:
    CAP#1 extends Serializable from capture of ?
    CAP#2 extends Serializable from capture of ?
    CAP#3 extends Serializable from capture of ?
    CAP#4 extends Serializable from capture of ?```

I've added a comment explaining this.